File: src/main/java/org/jukeboxmc/event/player/PlayerLoginEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerLoginEvent extends PlayerEvent{

    private String kickReason = "Disconnected";
    private Result result = Result.ALLOWED;

    /**
     * Creates a new {@link PlayerEvent}
     *
     * @param player who represents the player which comes with this event
     */
    public PlayerLoginEvent( Player player ) {
        super( player );
    }

    public String getKickReason() {
        return this.kickReason;
    }

    public void setKickReason( String kickReason ) {
        this.kickReason = kickReason;
    }

    public void setResult( Result result ) {
        this.result = result;
    }

    public Result getResult() {
        return this.result;
    }

    public static enum Result {
        ALLOWED,
        SERVER_FULL,
        WHITELIST,
        XBOX_AUTHENTICATED,
        OTHER
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerLoginEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.player.Player

class PlayerLoginEvent(player: Player) : PlayerEvent(player) {

    private var reason = "Disconnected"
    private var result = Result.ALLOWED

    fun getKickReason(): String {
        return this.reason
    }

    fun setKickReason(reason: String){
        this.reason = reason
    }

    fun setResult(result: Result) {
        this.result = result
    }

    fun getResult(): Result {
        return result
    }

    enum class Result {
        ALLOWED,
        SERVER_FULL,
        WHITELIST,
        XBOX_AUTHENTICATED,
        OTHER
    }
}